krogowsk531/house-rules
# Features: render home page & single game page

# Team members present: Hanna, Katy, Michael

## What's this PR do? 
* Refactor `Home` component to be a functional component that gets props using `getStaticProps`
* Add function to fetch single game
* Use `getStaticPaths` and `getStaticProps` to set `GameDetails` props dynamically for different games' details pages to render.
* Check functionality of `NavBar` using `Sample` page - this should get replaced with routing back to `Home` on click
## Where should the reviewer start? 
Main changes to:
`pages/index.js`
`pages/api/apiCalls.js`
`pages/GameDetails/[gameid].js`

## How should this be manually tested? 
Run `npm run dev` in terminal and open localhost:3000 in your browser. Should be able to click on each `SingleCard` element and be routed to `GameDetail` page

## Any background context you want to provide? 
We haven't done any testing yet so we should get that done before moving forward with more features. Currently using hardcoded data for `Home` fetch as to not overdo the 60 API calls/minute maximum set by the API we are using.

## What are the relevant tickets? 
* Addresses:
* Closes #1;
* Closes #2;
* Closes #7;
* Closes #9;
* Closes #16;
* Closes #17;

## Screenshots (if appropriate) 
n/a

## Questions:
* Can't seem to make it so that a `GameDetails` page can bring in game data as an object instead of object in an array inside an object by setting `props.game` to evaluate to a variable that evaluates to game.games[0] in `getStaticProps` so it's messier in the render
* Indenting in some of the files are spacing 4 instead of 2 -- how to fix?
